openlibrary/plugins/upstream/tests/test_addbook.py

```diff
@@ -348,3 +351,98 @@ def mock_user():
         assert new_work.title == "Original Edition Title"
         # Should ignore edits to work data
         assert web.ctx.site.get("/works/OL100W").title == "Original Work Title"
+
+
+class TestAddBook:
+    def setup_method(self, method):
+        web.ctx.site = MockSite()
+
+    def test_unpermitted_logged_in_user_cannot_add_book(self, monkeypatch):
+        _nonlocal = {'render_template_permission_denied': False}
```

Comment:
Discussed use of name `_nonlocal` with @cclauss and felt it was acceptable since this can be replaced with the built in `nonlocal` mechanism in python 3.

Reply:
Nitpick, but shouldn't this be `nonlocal_` to avoid name collision rather than a pre-underscore to imply internal use?